src/core/utils/platform.spec.ts

```diff
@@ -0,0 +1,76 @@
+import { getPlatform } from "./platform";
+import os from "os";
+
+describe("getPlatform", () => {
+  it("should call os.platform() when getPlatform is called", () => {
```

Comment:
Rather than having a separate method which is checking the mock invocation count.. shall we put the check in all the test methods?


```ts
  it("should return 'win-x64' when running on Windows", () => {
    // Mock the os.platform function to return "win32"
    const mockOSPlatform = jest.spyOn(os, "platform").mockReturnValue("win32");

    // Call the function and expect it to return "win-x64"
    expect(getPlatform()).toEqual("win-x64");
    expect(mockOSPlatform).toHaveBeenCalledTimes(1);
  });

```

Reply:
done